File: post_to_database.py
# This file is the new PostToDatabase.js

from __future__ import print_function
import boto3
import random
import json
import decimal
from boto3.dynamodb.conditions import Key, Attr

def check_item_existence(food_item):
    # Check the item exists

    dynamodb = boto3.resource("dynamodb")

    table = dynamodb.Table('Food_Items')
    
    response = table.query(
    KeyConditionExpression=Key('Identifier').eq(food_item[0])
    )

    print("check_item_existence called", food_item[0])

    if(len(response['Items']) == 0):
        return False

    return True
    


def add_item_to_DB(food_item):
    # Add to Food_Items DB
    print("add_item_to_DB called", food_item[0])

    dynamodb = boto3.resource("dynamodb")

    table = dynamodb.Table('Food_Items')

    response = table.put_item(
       Item={
            'Identifier': food_item[0],
            'Sodium': randint(20,100),
            'Potasium': randint(120,180)
            }
    )

    print("PutItem succeeded:")

File: PostToDatabase.js
var AWS = require("aws-sdk");

var docClient = new AWS.DynamoDB.DocumentClient();

console.log("Adding item in the Databse, table Food_Items");

var returnValue = false;

module.exports = {

  checkItemExistence: function(foodItem) {
    
      var params = {
      TableName : "Food_Items",
      KeyConditionExpression: "#id = :name",
      ExpressionAttributeNames:{
          "#id": "Identifier"
      },
      ExpressionAttributeValues: {
          ":name":foodItem.name
      }
  };

      docClient.query(params, function(err, data) {
          if (err) {
              console.error("Unable to query.");
          } 
          else {
              console.log("Query succeeded.");
              if (data.Items.length == 0)
              {
                returnValue = false;
              }
              else
              {
                returnValue = true;
              }
          }
      });
      console.log("We are getting inside checkItemExistence method");
      return returnValue;
  },


  addItemToDB: function(foodItem) {
    var params = {
        TableName:"Food_Items",
        Item: {
            "Identifier":  foodItem.name,
            "Sodium": (Math.floor(Math.random() * (100 - 20 + 1)) + 20),
            "Potasium":  (Math.floor(Math.random() * (250 - 120 + 1)) + 120)
        }
    };

    console.log("Adding a new item...");
    docClient.put(params, function(err, data) {
       if (err) {
           console.error("Unable to add food", foodItem.name);
       } else {
           console.log("PutItem succeeded:", foodItem.name);
       }
    });
    
    console.log("We are getting inside addItemToDB method");

  }

};


